### ISSUE TITLE
Race

### ISSUE THREAD

**Issue by NormandoHall** [Hexagon/croner#217]
**Describe the bug**
After a week refactoring a lot of code to fix an issue, I found a croner bug. 

**To Reproduce**

1. Set a cron task to schedule every 2 seconds
2. The cron triggers every 2 second, and eventually trigger after 3 o 4 milliseconds, causing a lot of issues in the program code. 
3. Leave running for at least few hours

```js
const {Cron} = require('croner');
let start = Date.now()

Cron('*/2 * * * * *', () => {
  const milis = Date.now() - start;
  if(milis < 1900) console.log(milis)
  start = Date.now()
});
```

**Expected behavior**
Respect run the job every 2 seconds

**System:**
 - OS: Linux
 - Runtime Node.js 18 and 16
 - Version croner 7.0.3 but seems this issue comes from 7.0.0 or before, I'm not sure.

**Additional context**
Testing cronner with the example of [overrun protection](https://croner.56k.guru/usage/examples/#overrun-protection) get a race condition also:

```
start
Call at 2023-10-20T22:23:46.007Z were blocked by call started at 2023-10-20T22:23:45.003Z
Call at 2023-10-20T22:23:47.005Z were blocked by call started at 2023-10-20T22:23:45.003Z
Call at 2023-10-20T22:23:48.003Z were blocked by call started at 2023-10-20T22:23:45.003Z
Call at 2023-10-20T22:23:49.001Z were blocked by call started at 2023-10-20T22:23:45.003Z
end
start
Call at 2023-10-20T22:23:50.003Z were blocked by call started at 2023-10-20T22:23:49.998Z
Call at 2023-10-20T22:23:50.005Z were blocked by call started at 2023-10-20T22:23:49.998Z
Call at 2023-10-20T22:23:50.007Z were blocked by call started at 2023-10-20T22:23:49.998Z
Call at 2023-10-20T22:23:50.010Z were blocked by call started at 2023-10-20T22:23:49.998Z
Call at 2023-10-20T22:23:50.012Z were blocked by call started at 2023-10-20T22:23:49.998Z
Call at 2023-10-20T22:23:50.014Z were blocked by call started at 2023-10-20T22:23:49.998Z
Call at 2023-10-20T22:23:50.016Z were blocked by call started at 2023-10-20T22:23:49.998Z
Call at 2023-10-20T22:23:50.018Z were blocked by call started at 2023-10-20T22:23:49.998Z
Call at 2023-10-20T22:23:50.020Z were blocked by call started at 2023-10-20T22:23:49.998Z
Call at 2023-10-20T22:23:50.024Z were blocked by call started at 2023-10-20T22:23:49.998Z
Call at 2023-10-20T22:23:50.026Z were blocked by call started at 2023-10-20T22:23:49.998Z
Call at 2023-10-20T22:23:50.028Z were blocked by call started at 2023-10-20T22:23:49.998Z
Call at 2023-10-20T22:23:50.030Z were blocked by call started at 2023-10-20T22:23:49.998Z
Call at 2023-10-20T22:23:50.032Z were blocked by call started at 2023-10-20T22:23:49.998Z
Call at 2023-10-20T22:23:50.034Z were blocked by call started at 2023-10-20T22:23:49.998Z
Call at 2023-10-20T22:23:50.036Z were blocked by call started at 2023-10-20T22:23:49.998Z
Call at 2023-10-20T22:23:50.038Z were blocked by call started at 2023-10-20T22:23:49.998Z
Call at 2023-10-20T22:23:50.040Z were blocked by call started at 2023-10-20T22:23:49.998Z
Call at 2023-10-20T22:23:50.043Z were blocked by call started at 2023-10-20T22:23:49.998Z
Call at 2023-10-20T22:23:50.045Z were blocked by call started at 2023-10-20T22:23:49.998Z
Call at 2023-10-20T22:23:50.047Z were blocked by call started at 2023-10-20T22:23:49.998Z
Call at 2023-10-20T22:23:50.049Z were blocked by call started at 2023-10-20T22:23:49.998Z
Call at 2023-10-20T22:23:50.051Z were blocked by call started at 2023-10-20T22:23:49.998Z
Call at 2023-10-20T22:23:50.053Z were blocked by call started at 2023-10-20T22:23:49.998Z
Call at 2023-10-20T22:23:50.055Z were blocked by call started at 2023-10-20T22:23:49.998Z
Call at 2023-10-20T22:23:50.057Z were blocked by call started at 2023-10-20T22:23:49.998Z
Call at 2023-10-20T22:23:50.059Z were blocked by call started at 2023-10-20T22:23:49.998Z
Call at 2023-10-20T22:23:50.061Z were blocked by call started at 2023-10-20T22:23:49.998Z
Call at 2023-10-20T22:23:50.063Z were blocked by call started at 2023-10-20T22:23:49.998Z and so on...
```

**Comment by Hexagon** [comment]
Thanks, very interesting bug report!

This is fixed in `croner@7.0.4.-dev.0`, can you see if it resolves your problem?

**Comment by NormandoHall** [comment]
Hello @Hexagon 

There is no `croner@7.0.4.-dev.0` package to install right now

```
npm i croner@7.0.4.-dev.0
npm ERR! code ETARGET
npm ERR! notarget No matching version found for croner@7.0.4.-dev.0.
npm ERR! notarget In most cases you or one of your dependencies are requesting
npm ERR! notarget a package version that doesn't exist.
```

**Comment by NormandoHall** [comment]
Well, after 6 hours seems it is fixed. The lowest value was 1756ms. And there is no race in protected mode.

Thanks for croner!!!

**Comment by Hexagon** [comment]
Awesome! Thanks for figuring out and reporting this. Fix released in stable branch from `croner@7.0.4`.
